src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3644,16 +3644,30 @@ void addToStringMethod(MethodSymbol printMethod)
 
                 if (!memberSignatures.TryGetValue(targetMethod, out Symbol? existingToStringMethod))
                 {
-                    var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
-                    members.Add(toStringMethod);
+                    var baseToStringMethod = getBaseToStringMethod();
+                    if (baseToStringMethod is null || !baseToStringMethod.IsSealed)
+                    {
+                        var toStringMethod = new SynthesizedRecordToString(this, printMethod, memberOffset: members.Count, diagnostics);
+                        members.Add(toStringMethod);
+                    }
                 }
-                else
+
+                MethodSymbol? getBaseToStringMethod()
```

Comment:
There's probably an easier way to find the base `ToString()` implementation, but I didn't find it... I'm open to suggestion.

Reply:
`Compilation.GetRuntimeMember` and `WellKnownMembers.GetDescriptor` can help.

Reply:
I'll try using that, thanks

Reply:
Maybe I'm missing something, but the `WellKnownMember` enum doesn't seem to know about `Object.ToString`... How do I get a `MemberDescriptor` for it?

Reply:
@jcouv any idea how I could do this? Or do I just leave it as is?

Reply:
Couldn't find anything better. Seems like that's what we have to do.